Sources/XcodeExport/Model/XcodeExportExtensions.swift

```diff
@@ -0,0 +1,225 @@
+        return FileContents(destination: destination, data: data)
+    }
+
+}
+
+// MARK: XcodeAssetContents
+
+extension XcodeAssetContents {
+
+    func makeFileContents(to directory: URL) throws -> FileContents {
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = .prettyPrinted
+
+        let data = try encoder.encode(self)
+        let fileURL = URL(string: "Contents.json")!
+
+        return FileContents(
+            destination: Destination(directory: directory, file: fileURL),
+            data: data
+        )
+    }
+
+}
+
+// MARK: ImagePack
+
+extension ImagePack {
+
+    func packForXcode() -> ImagePack? {
+        var xcodeImagePack = self
+        xcodeImagePack.images = images.filter { $0.isValidForXcode(scale: $0.scale) }
+        return xcodeImagePack
+    }
+
+    func makeFileContents(to directory: URL, preservesVector: Bool?, appearance: Appearance? = nil) throws -> [FileContents] {
+        let properties = { () -> XcodeAssetContents.TemplateProperties? in
+            if let preservesVector = preservesVector {
+                return XcodeAssetContents.TemplateProperties(
+                    preservesVectorRepresentation: preservesVector ? true : nil
+                )
+            } else {
+                return nil
+            }
+        }()
+
+        return try packForXcode()
+            .flatMap { imagePack -> [FileContents] in
+                let name = imagePack.name
+                let dirURL = directory.appendingPathComponent("\(name).imageset")
+                
+                let assetsContents = imagePack.makeXcodeAssetContentsImageData(appearance: appearance)
+                let contentsFileContents = try XcodeAssetContents(
+                    images: assetsContents,
+                    properties: properties
+                ).makeFileContents(to: dirURL)
+                
+                let files = imagePack.makeImageFileContents(to: dirURL, appearance: appearance)
+                
+                return files + [contentsFileContents]
+            } ?? []
+    }
+
+    func makeImageFileContents(to directory: URL, appearance: Appearance? = nil) -> [FileContents] {
+        images.map { $0.makeFileContents(to: directory, appearance: appearance) }
+    }
+
+    func makeXcodeAssetContentsImageData(appearance: Appearance? = nil) -> [XcodeAssetContents.ImageData] {
+        images.map { $0.makeXcodeAssetContentsImageData(scale: $0.scale, appearance: appearance) }
+    }
+
+}
+
+// MARK: AssetPair
+
+extension AssetPair where AssetType == ImagePack {
+
+    func makeFileContents(to directory: URL, preservesVector: Bool?) throws -> [FileContents] {
+        let name = light.name
+        let dirURL = directory.appendingPathComponent("\(name).imageset")
+
+        let lightPack = light.packForXcode()
+        let darkPack = dark?.packForXcode()
+
+        let properties = { () -> XcodeAssetContents.TemplateProperties? in
```

Comment:
This closure appears twice in this file. On line 76 and here.

Reply:
You can simplify code from line 124 to 132:
```
let properties = preservesVector.map {
    XcodeAssetContents.TemplateProperties(
        preservesVectorRepresentation: $0 ? true : nil
    )
}
```

Reply:
👍